File: frontend/src/app/model/ProductSearchDto.ts
import { Brand } from './Brand';
import { BrandCheckBox } from './BrandCheckBox';
import { SellerCheckbox } from './SellerCheckbox';

export interface ProductSearchDto {
  id: number;

  name: string;

  alias: string;

  price: number;

  discountPrice: number;

  discountStart: Date;

  discountEnd: Date;

  mainImage: string;

  brand: BrandCheckBox;

  seller: SellerCheckbox;
}

File: backend/src/main/java/com/canada/aws/dto/ProductSearchDto.java
package com.canada.aws.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductSearchDto {
    private Integer id;

    private String name;

    private String alias;

    private float price;

    private float discountPrice;

    private Date discountStart;

    private Date discountEnd;

    private String mainImage;
}
